### PR TITLE
Feature/add excel output

### PR DESCRIPTION
**Pull request by Firsttryy**
## PR checklist

- [ ] dev-branch has been merged into local branch to resolve conflicts
- [ ] the application has been manually tested after merge
- [ ] another dev reviewed and approved
- [ ] if feature-Branch: Teams PO has approved (show via e.g. screenshots/screencapture/live demo)

### CODE REVIEW

File: planner/parser/ModelObjectParser.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            inputSet.push(new DataObjectReference(dataclasses.find(element => element.name === dataObjectReference.get('sourceRef')[0].dataclass.name), dataObjectReference.get('sourceRef')[0].states[0].name, false));
+        }
+        let outputSet = [];
+        for (let dataObjectReference of action.get('dataOutputAssociations')) {
+            outputSet.push(new DataObjectReference(dataclasses.find(element => element.name === dataObjectReference.get('targetRef').dataclass.name), dataObjectReference.get('targetRef').states[0].name, false));
+        }
+        actions.push(new Action(action.name, parseInt(action.duration), parseInt(action.NoP), roles.find(element => element.name === action.role.name), new IOSet(inputSet), new IOSet(outputSet)))
+    }
+
+    return new Planner(currentState, goal, actions);
```
Comment (Firsttryy) [inline planner/parser/ModelObjectParser.js:89]: resources not used yet?

File: planner/types/Resource.ts [hunk 2]
```diff
@@ -2,16 +2,20 @@ import {Role} from "./Role";
...
+    public constructor(name: string, roles: Role[] = [], capacity: number) {
         this.name = name;
-        this.role = role;
+        this.roles = roles;
         this.capacity = capacity;
     }
 
     public satisfies(role: Role | null, NoP: number): boolean {
-        return role === this.role && NoP <= this.capacity;
+        if (!role) {
```
Comment (Firsttryy) [inline planner/types/Resource.ts:15]: take capacity into account though? or we should define completely other behavior

File: planner/types/executionState/ExecutionDataObjectInstance.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    dataObjectInstance: DataObjectInstance;
+    state: string;
+
+    public constructor(dataObjectInstance: DataObjectInstance, state: string) {
+        this.dataObjectInstance = dataObjectInstance;
+        this.state = state;
+    }
+
+
+    public isArray() {
```
Comment (Firsttryy) [inline planner/types/executionState/ExecutionDataObjectInstance.ts]: we do not need that anymore, right?

File: app/lib/fragmentmodeler/FragmentModeler.js [hunk 4]
```diff
@@ -126,11 +126,12 @@ FragmentModeler.prototype.getDataObjectReferencesOfClass = function (clazz) {
 }
 
 FragmentModeler.prototype.getTasksWithRole = function (role) {
-    return this.get('elementRegistry').filter((element, gfx) =>
```
Comment (Franka-T) [inline app/lib/fragmentmodeler/FragmentModeler.js]: so

File: planner/types/Resource.ts (hunk 2 above)
Comment (Firsttryy) [reply planner/types/Resource.ts:15]: Behavior will be, that Anyone can do it without assigning a resource (or assign new Anyone resource

**Review by Noel-Bastubbe — APPROVED** [review]

**Pull request merged**